devices/xiaomi.js

```diff
@@ -1471,11 +1824,15 @@ module.exports = [
         zigbeeModel: ['lumi.motion.ac01'],
         model: 'RTCZCGQ11LM',
         vendor: 'Xiaomi',
-        description: 'Aqara presence detector FP1 (regions not supported for now)',
-        fromZigbee: [fz.aqara_opple],
-        toZigbee: [tz.RTCZCGQ11LM_presence, tz.RTCZCGQ11LM_monitoring_mode, tz.RTCZCGQ11LM_approach_distance,
-            tz.aqara_motion_sensitivity, tz.RTCZCGQ11LM_reset_nopresence_status],
-        exposes: [e.presence().withAccess(ea.STATE_GET),
+        description: 'Aqara presence detector FP1 (experimental region support)',
+        fromZigbee: [fz.aqara_opple, fzLocal.aqara_fp1_region_events],
```

Comment:
Since both converters respond to the same message, can these be merged with each other? (aqara_fp1_region_events into `aqara_opple`?

Reply:
Does it really make sense though? `aqara_opple` (or rather its `numericAttributes2Payload`) seems to be really bloated, one to rule them all 500-lines function which is rather hard to read and comprehend. This would make sense if `region_events` was a generic thing, but it's not, so leaving it as a separate code entity makes more sense to me (albeit, some pieces from `numericAttributes2Payload` could be reused, eg. the "unknown key" handling).

However, I won't insist on this approach. If you think it's better to merge it, I'll try to do that, probably by extracting current code into a separate function and somehow calling it in `numericAttributes2Payload`, as that would most likely be the best compromise with this approach.

So, what do you think?

Reply:
I agree, with you, so let's leave it as is.